### ISSUE TITLE
Alignak objects template detection and management

### ISSUE THREAD

**Issue by mohierf** [Alignak-monitoring/alignak#371]
Where as Nagios considers that a configuration element is a template when it finds a `name` property, Alignak considers that an object is a template if it is not registered (`register` property is set to `0`).

The problem is that a valid Nagios configuration is not parsed correctly by the Alignak arbiter and it needs to be modified for being used in Alignak.

See Nagios templates doc: https://assets.nagios.com/downloads/nagioscore/docs/nagioscore/3/en/objectinheritance.html

See Alignak source code:

```
    def is_tpl(self):
        """
        Check if this object is a template

        :return: True if is a template, else False
        :rtype: bool
        """
        return not getattr(self, "register", True)
```

**Comment by mohierf** [comment]
**First case**, this host configuration must be detected as a template:
```
define host{
    name   lan_switch-host
    use       generic-host
    register  0 
}
```
Because:
- `register` is defined as 0

**Second case**, this host configuration must also be detected as a template:
```
define host{
    name   lan_switch-host
    use       generic-host
}
```
Because:
- `register` is not defined as 0
- a `name` property exists, so it is a template
- no `host_name` property exists, so it is **not** an host

**Third case**, this host configuration must also be detected as a template **and** an host:
```
define host{
    name   lan_switch-host
    use       generic-host
    host_name new_host 
}
```
Because:
- a `name` property exists, so it is a template
- an `host_name` property exists, so it is **also** an host because `register` is not defined as 0

**Comment by mohierf** [comment]
**First case**, this service configuration must be detected as a template:
```
define service {
    name      svc_template
    use       generic-service
    register  0
}
```
Because:
- `register` is defined as 0

**Second case**, this service configuration must also be detected as a template:
```
define service {
    name      svc_template
    use       generic-service
}
```
Because:
- `register` is not defined as 0
- a `name` property exists, so it is a template
- no `host_name` nor `service_description`property exists, so it is **not** a service of an host

**Third case**, this service configuration must also be detected as a template **and** a service:
```
define service {
    name      svc_template
    use       generic-service
    host_name new_host 
    service_description new_service
}
```
Because:
- a `name` property exists, so it is a template
- an `host_name` and a `service_description` property exist, so it is **also** a service of an host because `register` is not defined as 0

**Comment by Seb-Solon** [comment]
I am not 100% sur we want to reimplement this Nagios behavior. If you want to have the possibility of having template and a host/service at the same time then create explicit parameter for each case that you can turn off/on. name is really counter intuitive. 

another thing, the fact of allowing this kind of configuration will make error detection more difficult. I am not sure that users are really willing to define a template and a host at the same time.

**Comment by mohierf** [comment]
I do agree with you this is why I did not made a PR with this ... 

I made some modifications to validate that it would be possible and not too tricky to find out hosts/templates in a Nagios-like configuration. I think that we should not change the Alignak configuration parser but rather raise alerts when we find out a definition of an host + template

This tricky Nagios feature is not that much used and it will be enough to make the user change its configuration when he migrates.

What about this? @ddurieux @Seb-Solon
